### DISCUSSION TITLE
Is that possible to query for the “raw” data within a time range with exact ingested time stamps without considering arbitrary step for range queries

### DISCUSSION THREAD

**Discussion by venkatchinmay** [prometheus/prometheus#10503] [Q&A] [answered]
Hi, I would like to retrieve data AS it was collected and recorded by Prometheus.
I do not want to use an arbitrary step for range queries.

**Is that possible to query for the “raw” data within a time range with exact ingested time stamps ?
If it is possible to get, so how ?
if it is not possible why ? and suggest any alternative**

@codesome

**Comment by roidelapluie** [comment, marked as answer]
yes, it is possible.

You need to use an **instant query** and run: `my_metric[1h]`
Due to the prometheus data model, it is not safe to graph that data because we do not output staleness markers. Therefore it is only available in console tab of Prometheus. While it is possible to visualize this in grafana, it is really not recommended and should not be done because grafana expects prometheus input to come in predefined query_range steps.

**Comment by venkatchinmay** [comment]
Thanks for your reply @roidelapluie @codesome
